submission/races.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package races
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"reflect"
+	"sync"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/ctpolicy"
+)
+
+const (
+	// URLProtocol keeps protocol substring for Log URLs.
+	URLProtocol = "https://"
+
+	// Sleeptim is duration between parallel batch call and subsequent requests to Logs within group.
+	SleepTime = time.Second
+)
+
+type guardedLogs struct {
+	logURLs map[string]bool
+	m       sync.RWMutex
+}
+
+func (gList *guardedLogs) add(logURL string) {
+	gList.m.Lock()
+	gList.logURLs[logURL] = true
+	gList.m.Unlock()
+}
+
+func (gList *guardedLogs) has(logURL string) bool {
+	gList.m.RLock()
+	exist := gList.logURLs[logURL]
+	gList.m.RUnlock()
+	return exist
+}
+
+func (gList *guardedLogs) sumFor(URLs map[string]bool) int {
+	var entries int
+	gList.m.RLock()
+	for url := range URLs {
+		if gList.logURLs[url] {
+			entries++
+		}
+	}
+	gList.m.RUnlock()
+	return entries
+}
+
+type result struct {
```

Comment:
maybe `submissionResult` ?

Reply:
?

Reply:
`result` is very generic -- result of what?  A terse name is fine if the scope of use of the type is short, but as the scope gets bigger the name needs to be more descriptive.  This type is package-private but is passed between goroutines, so is somewhere in the middle.